src/com/carolinarollergirls/scoreboard/jetty/WS.java

```diff
@@ -266,7 +265,54 @@ private void sendWSUpdatesForPaths(Set<String> watchedPaths, Set<String> changed
         }
 
         protected UUID id;
-        protected Set<String> paths = new HashSet<>();
+        protected PathTrie paths = new PathTrie();
         private Map<String, Object> state;
+
+        private class PathTrie {
+          boolean exists = false;
+          Map<Character, PathTrie> trie = new HashMap<>();
+
+          public void addAll(Set<String> c) {
+            for (String p: c) {
+              add(p);
+            }
+          }
+          public void add(String p) {
+            PathTrie head = this;
+            for (int i = 0;; i++) {
+              if (head.exists) {
+                // Already covered.
+                return;
+              }
+              if (i >= p.length()) {
+                break;
+              }
+              if (head.trie.containsKey(p.charAt(i))) {
+                 head = head.trie.get(p.charAt(i));
+              } else {
+                 PathTrie child = new PathTrie();
+                 head.trie.put(p.charAt(i), child);
+                 head = child;
+              }
+            }
+            head.exists = true;
+          }
+          public boolean covers(String p) {
+            PathTrie head = this;
+            for (int i = 0;; i++) {
+              if (head.exists) {
+                // Already covered.
```

Comment:
This comment doesn't really make sense in this function.

Reply:
Removed